util/system.go

```diff
@@ -61,9 +61,14 @@ func GetPort(low, hi int) int {
 }
 
 func PortAvailable(port string) bool {
+	ln, err := net.Listen("tcp", "localhost:"+port)
+	if err != nil {
+		return false
+	}
 
-	ln, err := net.Listen("tcp", ":"+port)
+	ln.Close()
 
+	ln, err = net.Listen("tcp", ":"+port)
```

Comment:
why two checks? Ideally we would test the port we are supposed to be listening in

Reply:
I had to do is this way because for some reason it didn't catch a busy port that was taken either by hasura cli (non docker) or some service in docker

Reply:
that's because you weren't checking the right socket. It's not the same to listen on `:$PORT` that to listen on `$IP:$PORT`. If you have multiple IPs you can even have different services listening on the same port on different IPs. So this code is checking more than it should.

Reply:
Can you give an example of a working code for that case?